### PR TITLE
Am/chore/next tfhe wip

### PR DESCRIPTION
**Pull request by IceTDrinker**
### PR content/description

Various improvements to the build system to be able to use the crate as a dependency in a semver-trick setting, main change is a way to by pass the build.rs script which was erroring due to weird cargo conflicts on package names (and maybe the fact our build.rs is not that smart to begin with with path allocations and such, so being able to skip it entirely just makes things easier for us for the semver trick)

The added TFHE_SPEC in the Makefile is to make sure Cargo can disambiguate the fact we are talking about our package and not a dependency, see https://github.com/rust-lang/cargo/issues/12891

### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,6 +17,10 @@ FAST_TESTS?=FALSE
 FAST_BENCH?=FALSE
 BENCH_OP_FLAVOR?=DEFAULT
 NODE_VERSION=20
+TFHE_CURRENT_VERSION:=$(shell grep '^version[[:space:]]*=' tfhe/Cargo.toml | cut -d '=' -f 2 | xargs)
```
Comment (mayeul-zama) [inline Makefile:20]: This could give the wrong result if a crate dependency is defined before the package version and its version is in new line 

That's unlikely

File: tfhe/cbindgen.toml [hunk 2]
```diff
@@ -114,14 +114,15 @@ bitflags = false
 ############## Options for How Your Rust library Should Be Parsed ##############
 
 [parse]
-parse_deps = true
-include = ["tfhe"]
+parse_deps = false
+include = []
```
Comment (mayeul-zama) [inline tfhe/cbindgen.toml:118]: What does this change? Why is this necessary?

File: tfhe/build.rs [hunk 3]
```diff
@@ -63,8 +64,8 @@ fn gen_c_api() {
     };
 
     cbindgen::Builder::new()
-        .with_crate(crate_dir.clone())
-        .with_config(cbindgen::Config::from_root_or_default(crate_dir))
+        .with_crate(crate_dir.as_path())
+        .with_config(cbindgen::Config::from_file(crate_dir.join("cbindgen.toml")).unwrap())
```
Comment (mayeul-zama) [inline tfhe/build.rs:68]: Is this to be more strict by precaution?

File: tfhe/cbindgen.toml (hunk 2 above)
Comment (IceTDrinker) [reply tfhe/cbindgen.toml:118]: it was a mistake from the time the C API required macro expansion, I think I made the mistake at the time, we don't need to parse deps, only our own crate, the include here is not required as it refers to dependencies as far as I understand the cbindgen doc, the parse expand is handled in the build.rs to programatically feed features that are enabled

File: tfhe/build.rs (hunk 3 above)
Comment (IceTDrinker) [reply tfhe/build.rs:68]: yes

File: Makefile (hunk 1 above)
Comment (mayeul-zama) [reply Makefile:20]: Not sure if it exist, but inserting a utility which asserts that a stream is only one line and forwards it would be more secure (even if not perfect)

Comment (IceTDrinker) [reply Makefile:20]: though it looks like the Cargo.toml format does not allow new lines for { version = "..." } stuff

Comment (mayeul-zama) [reply Makefile:20]: I think you can have 
```
[dependencies.dep_name]
version = "1.0"
```

**Comment by IceTDrinker** [comment]
gotta love that the unambiguous spec for run (and only run) does not work

File: Makefile (hunk 1 above)
Comment (IceTDrinker) [reply Makefile:20]: yeah :|

Comment (IceTDrinker) [reply Makefile:20]: I guess I could trim the file once we find the word "dependencies"

Comment (IceTDrinker) [reply Makefile:20]: something like this maybe https://stackoverflow.com/questions/7103531/how-to-get-the-part-of-a-file-after-the-first-line-that-matches-a-regular-expres

Comment (IceTDrinker) [reply Makefile:20]: or maybe a light toml parser

Comment (mayeul-zama) [reply Makefile:20]: That seems overkill to me

Comment (IceTDrinker) [reply Makefile]: got something a bit more robust

**Comment by IceTDrinker** [comment]
> looks ok to me

Approve if you think it can go for merging pretty please 😄

**Review by tmontaigu — APPROVED** [review]
